File: InterviewCakeProblems2/reverse_string_in_place.rb
def reverse_string_in_place(char_arr)
    start = 0
    endIdx = char_arr.length - 1

    while start < endIdx
        temp = char_arr[start]
        char_arr[start] = char_arr[endIdx]
        char_arr[endIdx] = temp;
        start += 1
        endIdx -= 1
    end
    char_arr
end

p reverse_string_in_place(['a','b','c','d'])
p reverse_string_in_place(['j','o','n','n','a','t','a','n'])
p reverse_string_in_place(['j','o','b'])

File: InterviewCakeProblems2/reverseStringInPlace.js
function reverseStringInPlace(charArr) {
    if (charArr.length < 2) {
        throw new Error("input must have at least 2 chars");
    }

    let start = 0;
    let end = charArr.length - 1;

    while (start < end) {
        let temp = charArr[start];
        charArr[start] = charArr[end];
        charArr[end] = temp
        start++;
        end--;
    }
    return charArr;
}

console.log(reverseStringInPlace(['a','b','c','d']))
console.log(reverseStringInPlace(['j','o','n','n','a','t','a','n']))
console.log(reverseStringInPlace(['j','o','b']))

